dilqr_rs/dilqr_rs_gradient.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+from dilqr_rs.dilqr import DiLQR
+
+class DiLQR_RS_Gradient(DiLQR):
```

Comment:
`DiLqrSampledGradient`?

Reply:
`D` is Direct, `i` is iterative, and `RS` is randomized smoothing :) (Again, bit of a naming crisis here....)

On the first part, how about `DirectIterativeLqr`? There are lots of abbreviations here which makes it a bit awkward to stick to the coding convention, but I think this is the best shot.  

For the second part, I agree we can choose between `Sampling` and `Random`. `GradientSampling` or `RandomizedGradient` maybe? (just to be consistent with the gradient sampling algorithm) - similarly we can do `ZeroOrderSampling`. 

This way we have 
`DirectIterativeLqrExact`, `DirectIterativeLqrGradientSampling`, and `DirectIterativeLqrZeroOrderSampling`. 
but feels like a mouthful!

Reply:
I think it's also worthwhile to take our time to think about good names on this one for marketing (separate from coding names). 

Maybe randomized iLQR sounds more catchy and direct is just a detail we can leave out?